docs/apache-airflow/howto/dynamic-dag-generation.rst

```diff
@@ -140,3 +140,74 @@ Each of them can run separately with related configuration
 
 .. warning::
   Using this practice, pay attention to "late binding" behaviour in Python loops. See `that GitHub discussion <https://github.com/apache/airflow/discussions/21278#discussioncomment-2103559>`_ for more details
+
+
+Optimizing DAG parsing delays during execution
+----------------------------------------------
+
+Sometimes when you generate a lot of Dynamic DAGs from a single DAG file, it might cause unnecessary delays
+when the DAG file is parsed during task execution. This is visible as delay before a task starts
+and when it gets executed. Sometimes the delays might be minutes.
+
+Why is this happening ?
+
+You might not be aware but when your task is executed just before execution, airflow parses the Python
+file the dag comes from. When tasks are executed, parsing time of dynamically generated DAGs
+when task are executed can be optimized. This optimization is most effective when the
+number of generated DAGs is high.
+
+The Airflow Scheduler (or DAG Processor) requires loading of a complete DAG file to process all metadata.
+However, task execution requires only a single DAG object to execute a task. Knowing this, allows us to
+skip the generation of unnecessary DAG objects when task is executed, shortening the parsing time.
+Upon evaluation of a DAG file, command line arguments are supplied which we can use to determine which
+airflow component performs parsing:
+
+* Scheduler/DAG Processor args: ``["scheduler"]`` or ``["dag-processor"]``
+* Task execution args: ``["airflow", "tasks", "run", "dag_id", "task_id", ...]``
+
+However, depending on the executor used and forking model, those args might be available via ``sys.args``
+or via name of the process running. Airflow either executes tasks via running a new Python interpreter or
+sets the name of the process as "airflow task supervisor: {ARGS}".
+
+Upon iterating over the collection of things to generate DAGs for, you can use these arguments to determine
+whether you need to generate all DAG objects (when parsing in the DAG File processor), or to generate only
+a single DAG object (when executing the task).
+
+
+.. code-block:: python
+  :emphasize-lines: 1,2,3,7,8
+
+  import sys
+  import ast
+
+  current_dag = None
+  if len(sys.argv) > 3 and sys.argv[1] == "tasks":
+      current_dag = sys.argv[3]
+  else:
+      try:
+          PROCTITLE_PREFIX = "airflow task supervisor: "
+          proctitle = str(setproctitle.getproctitle())
+          if proctitle.startswith(PROCTITLE_PREFIX):
+              args_string = proctitle[len(PROCTITLE_PREFIX) :]
+              args = ast.literal_eval(args_string)
+              if len(args) > 3 and args[1] == "tasks":
+                  current_dag = args[3]
+      except:
+          pass
```

Comment:
IMO this if/else could use a comment e.g. "args supplied to process" and "process is forked and task args are supplied via proctitle".

Reply:
Yeah. Comments are needed but rather than repeating in the comment what we actually do, it should explain why we are doing it:

```
# task executed by starting a new Python interpreter
```
and

```
 # task executed via forked process
````

are better comments in this case :)